gnovm/cmd/gno/mod.go

```diff
@@ -152,3 +171,89 @@ func execModInit(args []string) error {
 
 	return nil
 }
+
+func execModTidy(args []string, io *commands.IO) error {
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	fname := filepath.Join(wd, "gno.mod")
+	gm, err := gnomod.ParseGnoMod(fname)
+	if err != nil {
+		return err
+	}
+
+	// Drop all existing requires
+	for _, r := range gm.Require {
+		gm.DropRequire(r.Mod.Path)
+	}
+
+	imports, err := getGnoImports(wd)
+	if err != nil {
+		return err
+	}
+	for _, im := range imports {
+		// skip if importpath is modulepath
+		if im == gm.Module.Mod.Path {
+			continue
+		}
+		gm.AddRequire(im, "v0.0.0-latest")
+	}
+
+	gm.Write(fname)
+	return nil
+}
+
+// getGnoImports returns the list of gno imports from a given path.
+// Note: It ignores subdirs. Since right now we are still deciding on
+// how to handle subdirs.
+// See:
+// - https://github.com/gnolang/gno/issues/1024
+// - https://github.com/gnolang/gno/issues/852
+//
+// TODO: move this to better location.
+func getGnoImports(path string) ([]string, error) {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	allImports := make([]string, 0)
+	seen := make(map[string]struct{})
+	for _, e := range entries {
+		filename := e.Name()
+		if ext := filepath.Ext(filename); ext != ".gno" {
+			continue
+		}
+		if strings.HasSuffix(filename, "_filetest.gno") {
+			continue
+		}
+		data, err := os.ReadFile(filepath.Join(path, filename))
+		if err != nil {
+			return nil, err
+		}
+		fs := token.NewFileSet()
+		f, err := parser.ParseFile(fs, filename, data, parser.ImportsOnly)
+		if err != nil {
+			return nil, err
+		}
+		for _, imp := range f.Imports {
+			importPath := strings.TrimPrefix(strings.TrimSuffix(imp.Path.Value, `"`), `"`)
+			if !strings.HasPrefix(importPath, "gno.land/") {
```

Comment:
Umm. I'm sure there are many place where "gno.land" is hard coded.  

I'm okay with bth, applying your changes here or to wait till we find a good solution and modify everywhere in a separate PR.

WDYT?

Reply:
I think right now we should only have "gno.land/*" import paths, while understanding that with IBC/ICS this will change and will allow for more domains, so that we should still treat all import paths which have a dot (before the first /) to come from external packages.

In any case, whichever is fine here. I've made a note on #1299.